### PR TITLE
Taking a look at the changes that FlowSpec introduces

### PR DESCRIPTION
**Pull request by Apanatshka**
Maybe I should rebase on master. I wonder if that's where the spurious POM file changes come from.

### CODE REVIEW

File: org.metaborg.meta.nabl2.java/src/main/java/org/metaborg/meta/nabl2/solver/components/SymbolicSolver.java [hunk 1]
```diff
@@ -31,7 +31,7 @@ public SymbolicSolver(Solver solver) {
     }
 
     @Override protected Unit doAdd(ISymbolicConstraint constraint) throws UnsatisfiableException {
-        constraint.matchOrThrow(CheckedCases.of(fact -> {
+        constraint.<Unit, UnsatisfiableException>matchOrThrow(CheckedCases.of(fact -> {
```
Comment (Apanatshka) [inline org.metaborg.meta.nabl2.java/src/main/java/org/metaborg/meta/nabl2/solver/components/SymbolicSolver.java]: I don't know why, but I had to add these types to get the project to compile at all.

**Comment by hendrikvanantwerpen** [comment]
It's porbably a good idea to make sure you are up to date with master first. The development branch is still merging from master as well.

File: org.metaborg.meta.nabl2.java/src/main/java/org/metaborg/meta/nabl2/solver/components/SymbolicSolver.java (hunk 1 above)
Comment (hendrikvanantwerpen) [reply org.metaborg.meta.nabl2.java/src/main/java/org/metaborg/meta/nabl2/solver/components/SymbolicSolver.java]: Inference does not always work as you'd want. I had other places where I had to make them explicit as well.

**Comment by hendrikvanantwerpen** [comment]
Overall the changes seem pretty contained. Moving some code around should resolve most issues. However, the pcollections are removed as a dependency, and replaced by @msteindorfer capsule library. I've rewritten all my data structures to follow the same immutable/transient structure that the capsule datastructures have. So, you probably want to do the same, but if you use capsule internally, this is not very hard to do IME.

**Pull request closed without merging**